core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -118,17 +118,13 @@ public static void schedule(String threadName) {
   }
 
   /**
-   * Waits until the given thread can be executed.
+   * Waits for up to 10 seconds until the given thread can be executed.
    *
    * @param name a name of the thread to wait for
    * @throws InterruptedException if the waiting thread is interrupted
    */
   public static void await(String name) throws InterruptedException {
-    try (LockResource r = new LockResource(sLock)) {
-      while (!sTimers.containsKey(name)) {
-        sCondition.await();
-      }
-    }
+    await(name, 10, TimeUnit.SECONDS);
```

Comment:
there should always be a method that does not timeout ... this PR should provide two version of execute (one with timeout and one without, which determines which of the two await methods is used) ... the tests can then choose which execute method to use ... this will result in more readable code because it is clear from the test that the execute method may (or may not) timeout